tests/Unit/BucketTesting/StreamBucketLoggerTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Tests\Unit\BucketTesting;
+
+use PHPUnit\Framework\TestCase;
+use WMDE\Fundraising\Frontend\BucketTesting\Bucket;
+use WMDE\Fundraising\Frontend\BucketTesting\Campaign;
+use WMDE\Fundraising\Frontend\BucketTesting\Logging\StreamBucketLogger;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\FakeBucketLoggingEvent;
+
+/**
+ * @covers \WMDE\Fundraising\Frontend\BucketTesting\Logging\StreamBucketLogger
+ */
+class StreamBucketLoggerTest extends TestCase {
+
+	public function testLogWriterAddsDate() {
+		$logfile = fopen( 'php://memory', 'a+' );
+		$logWriter = new StreamBucketLogger( $logfile );
+		$logWriter->setDateFormat( 'Y-m-d H:i' ); // Ignore seconds to avoid Heisentests
```

Comment:
haha :)

Isn't the issue still there though? Just 60 times less likely to happen?

Reply:
The execution time where this is relevant is just a few milliseconds (maybe even less), so the odds of this being a problem is like a 1 in 60000 chance which should be good enough, otherwise we should start considering ditching Wikipedia and start playing the lottery.

Reply:
No need to have this chance at all though